### DISCUSSION TITLE
How to use chatComposite and chat Adapter for teams interop?

### DISCUSSION THREAD

**Discussion by anonymous21155** [Azure/communication-ui-library#5465] [General]
I have been using both callComposite and callWithChatComposite for teams interoperability. However I was not able use this with chatComposite and chatAdapter. I can see from the [doc](https://azure.github.io/communication-ui-library/?path=/docs/composites-adapters--docs) locator is not a property for ```AzureCommunicationChatAdapterArgs``` while its supported for ```AzureCommunicationCallAdapterArgs.``` Is there a way i can use teams interoperability with chatComposite?

**Comment by Leah-Xia-Microsoft** [comment]
Hi @anonymous21155,

To clarify, are you looking to enable Teams interop Chat **without** a Teams meeting? If so, please note that we currently only support Teams Interop for meeting chats, where the participant must first join via Calling SDK and be admitted to the meeting.

If your goal is to enable Teams interop Chat in the context of a Teams meeting, since the chatComposite is designed without calling functionality, it's not intended to handle the process of participants joining a call and requesting admission, whereas the callWithChatComposite includes this flow as part of its design.

If the user is already added and admitted to the meeting, then the user can have a Teams interop chat using the ChatComposite. 

If you have a specific use case, please submit a feature request through [this link](https://feedback.azure.com/d365community/forum/81ff6d2b-0c25-ec11-b6e6-000d3a4f0858). Thanks.

**Reply by anonymous21155** [reply]
Hi @Leah-Xia-Microsoft, 
Thank you for your resposnse. I am trying Teams interop with a Teams meeting itself. Since locator is not a property for chatComposite, I am not able to complete the interoperability. Could it be possible to bring this in the future release?

![image](https://github.com/user-attachments/assets/020dfd8c-e182-401c-8881-952ed770756b)

**Reply by Leah-Xia-Microsoft** [reply]
Hi @anonymous21155,

Thank you for providing more details about your scenario.

The ChatComposite is designed to support Teams interop meeting chats only after participants have joined and been admitted to the meeting.

You can implement the logic to add and admit a participant to a meeting by following this guide: [Quickstart: Join your chat app to a Teams meeting](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/chat/meeting-interop?pivots=platform-web).

Once this is set up, you can pass the [threadId](https://learn.microsoft.com/en-us/javascript/api/azure-communication-services/@azure/communication-calling/callinfo?view=azure-communication-services-js#@azure-communication-calling-callinfo-threadid) (along with other required properties) to the ChatComposite, and the Teams interop meeting chat will function as expected.

Please let us know if you have any further questions. Thanks!
